doc/builders/trivial-builders.chapter.md

```diff
@@ -31,34 +46,53 @@ An example of using `runCommand` is provided below.
+To minimize your dependencies, you should only use this if you are sure you will need a C compiler to run your command.
 
 ## `runCommandLocal` {#trivial-builder-runCommandLocal}
 
-Variant of `runCommand` that forces the derivation to be built locally, it is not substituted. This is intended for very cheap commands (<1s execution time). It saves on the network round-trip and can speed up a build.
+A variant of `runCommand` that ensures the derivation is built locally instead of substituting a remote build.
+It can save time and speed up a larger collection of builds by avoiding the network round-trip to check if a remote substitution is available.
+However, it is only intended for very cheap commands (<1s execution time).
 
 ::: {.note}
-This sets [`allowSubstitutes` to `false`](https://nixos.org/nix/manual/#adv-attr-allowSubstitutes), so only use `runCommandLocal` if you are certain the user will always have a builder for the `system` of the derivation. This should be true for most trivial use cases (e.g., just copying some files to a different location or adding symlinks) because there the `system` is usually the same as `builtins.currentSystem`.
+This sets `allowSubstitutes` to `false`.
+Only use `runCommandLocal` if you are certain the user will always have a builder for the `system` of the derivation.
+This should be true for most trivial use cases (e.g., just copying some files to a different location or adding symlinks)
+because there the `system` is usually the same as `builtins.currentSystem`.
 :::
 
-## `writeTextFile`, `writeText`, `writeTextDir`, `writeScript`, `writeScriptBin` {#trivial-builder-writeText}
+## `writeTextFile` {#trivial-builder-writeTextFile}
+
+There are a range of functions that let you write text files into the Nix store, starting with `writeTextFile`.
+
+It accepts one argument as an attribute set containing the following names:
 
-These functions write `text` to the Nix store. This is useful for creating scripts from Nix expressions. `writeTextFile` takes an attribute set and expects two arguments, `name` and `text`. `name` corresponds to the name used in the Nix store path. `text` will be the contents of the file. You can also set `executable` to true to make this file have the executable bit set.
+* `name` is the name that Nix will append to the store path in the same way that `stdenv.mkDerivation` uses its `name` attribute.
+* `text` is the contents of the file to be written.
+* `executable` is an optional boolean flag indicating if the output file should be executable. It is `false` by default.
+* `destination` is an optional relative path to be appended to the store path when adding the file to the Nix store.
+* `checkPhase` is an optional script that can be run to verify the output file. It is empty by default and is typically used to perform syntax checks.
+* `allowSubstitutes` is an optional boolean flag indicating if the derivation can be substituted for a remote build.
```

Comment:
Not entirely right, a remote build is not a substitution. Let's link to https://nixos.org/manual/nix/stable/language/advanced-attributes.html?highlight=allowSubstitut#adv-attr-allowSubstitutes instead. Also https://nixos.org/manual/nix/stable/glossary#gloss-substitute